test/controllers/sessions_controller_test.rb

```diff
@@ -196,6 +210,79 @@ class SessionsControllerTest < ActionController::TestCase
       end
     end
 
+    ["my-shop", "my-shop.myshopify.com", "https://my-shop.myshopify.com",
+     "http://my-shop.myshopify.com",].each do |good_url|
```

Comment:
Absolute nit, and completely my personal preference, so I apologize in advance for being pedantic, but what do you think about listing one of these per line? Makes it easier to grok that it's multiple test cases IMO.